enhancements/machine-config/user-data-secret-managed.md

```diff
@@ -0,0 +1,113 @@
+  - "/enhancements/that-less-than-great-idea.md"
+superseded-by:
+  - "/enhancements/our-past-effort.md"
+---
+
+# Machines user-data managed by the MCO
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+
+## Summary
+
+Currently the installer creates [pointer Ignition configs][1] which end up as [user][2] [data][3] secrets [referenced by the installer-generated Machine(Sets)][4].  That makes things like the Ignition v2 (spec 3) transition tricky.
+The key point is that the Secret holding the initial stub ignition is completely unmanaged after it's created by the installer.
+In order to upgrade to the new Ignition spec, the MCO needs a way to update the Secret to the new Ignition spec.
+
+
+[1]: https://github.com/openshift/installer/blob/093ca65398fe567bdf63322894496cebbe3d2ade/pkg/asset/ignition/machine/node.go#L30-L36
+[2]: https://github.com/openshift/installer/blob/093ca65398fe567bdf63322894496cebbe3d2ade/pkg/asset/machines/master.go#L161-L170
+[3]: https://github.com/openshift/installer/blob/093ca65398fe567bdf63322894496cebbe3d2ade/pkg/asset/machines/worker.go#L197-L205
+[4]: https://github.com/openshift/installer/blame/093ca65398fe567bdf63322894496cebbe3d2ade/docs/user/aws/install_upi.md#L231-L232
+
+## Motivation
+
+The goal for this enhancement is to have the Secret that contains the stub ignition configuration managed by the MCO. The MCO will later be able to update the stub to a newer version of the spec during an upgrade.
+
+### Goals
+
+The MCO owns the lifecycle of the Secret containing the pointer config. The installer is still responsible to create the initial version of the Secret as that is required to support UPI installation by generating the initial ignition configs until the installer provisions the control plane via MAO and not TF. The MCO takes ownership of the Secret containing the stub ignition once the cluster is up and running. When we'll update to the new Ignition version, the MCO can update the Secret.
+
+The final goal for this enhancement is to be able to manage that Secret from the MCO during the ignition migration to the new spec.
+
+If we don't manage that Secret, its content will be on an unsupported version of Ignition that can't provision new machines joining the cluster.
+
+### Non-Goals
+
+To make this simple and to be able to fit this into the release that will bring Ignition v2, we're not considering per-machine Secret(s) as highlighted here https://github.com/openshift/machine-config-operator/issues/683#issuecomment-493193902 and here https://github.com/openshift/machine-config-operator/issues/683#issuecomment-640602493
+
+We don't want to diverge from the current implementation where the Installer still creates the Secret and stub ignition on installation otherwise we'll have to change far more things in order to keep supporting UPI installation that needs ignition configs before starting to install.
```

Comment:
as per https://github.com/openshift/enhancements/pull/368/files#r440041955 couldn't the mco gain the ability to render the secrets when running from disk?

Reply:
we can 100% do that but today, since we need the raw ignition for UPI and as you said, TF we still need the installer to run early enough and the MCO is too late to the party

Reply:
UPI customers don't have a place to render the pointer ignitions using the mco operator and therefore we need to continue to support the adoption.